src/test/java/seedu/address/model/person/PhoneTest.java

```diff
@@ -15,11 +15,17 @@ public void isValidPhone() {
         assertFalse(Phone.isValidPhone("91")); // less than 3 numbers
         assertFalse(Phone.isValidPhone("phone")); // non-numeric
         assertFalse(Phone.isValidPhone("9011p041")); // alphabets within digits
-        assertFalse(Phone.isValidPhone("9312 1534")); // spaces within digits
+        assertFalse(Phone.isValidPhone("HIP: 9011041")); // Incorrect phone type
+        assertFalse(Phone.isValidPhone("HOP: 9011041 Oll: 9893898 Hm: 971897")); // Multiple incorrect phone type
 
         // valid phone numbers
         assertTrue(Phone.isValidPhone("911")); // exactly 3 numbers
         assertTrue(Phone.isValidPhone("93121534"));
         assertTrue(Phone.isValidPhone("124293842033123")); // long phone numbers
+        assertTrue(Phone.isValidPhone("93121234 15341234")); // spaces within digits
+        assertTrue(Phone.isValidPhone("HP: 928782711")); // hand phone number
+        assertTrue(Phone.isValidPhone("O: 9134343431")); // office number
+        assertTrue(Phone.isValidPhone("H: 911")); // house number
+        assertTrue(Phone.isValidPhone("O: 911 H: 972727 HP: 9234817")); // office number
```

Comment:
What if users enter their numbers in a different order?
Or maybe they did not label some phone numbers?
e.g."34567892 H:3214123 HP:12345678"

Reply:
They can enter their numbers in any order, as long there are at least 3 numbers. I give them the freedom to choose whether they want to include labels or not for more flexibility for the users, so 4567892 H:3214123 HP:12345678 will also work